VultisigApp/VultisigApp/Services/RpcEvmService.swift

```diff
@@ -253,46 +254,18 @@ class RpcEvmService: RpcService {
         }
     }
     
-    private var cacheTokens = ThreadSafeDictionary<String, (data: [Token], timestamp: Date)>()
-    func getTokens(urlString: String) async -> [Token] {
-        let cacheKey = urlString
-        let cacheDuration: TimeInterval = 60 * 10 // Cache duration of 10 minutes
-        
-        if let cachedTokens: [Token] = await Utils.getCachedData(cacheKey: cacheKey, cache: cacheTokens, timeInSeconds: cacheDuration) {
-            print("Returning tokens from cache")
-            return cachedTokens
-        }
+    func getTokens(nativeToken: Coin) async -> [CoinMeta] {
         
         do {
-            let data: Data = try await Utils.asyncGetRequest(urlString: urlString, headers: [:])
-            if var tokens: [Token] = Utils.extractResultFromJson(fromData: data, path: "tokens", type: Token.self, mustHaveFields: ["tokenInfo.website", "tokenInfo.image"]) {
-                tokens = tokens.map { token in
-                    var mutableToken = token
-                    if let image = mutableToken.tokenInfo.image{
-                        mutableToken.tokenInfo.setImage(image: "\(extractTokenDomainURL(from: urlString))\(image)" )
-                    }
-                    return mutableToken
-                }
-                cacheTokens.set(cacheKey, (data: tokens, timestamp: Date()))
-                return tokens
-            } else {
-                return []
+            let oneInchTokens = try await oneInchService.fetchTokensBalance(chainId: nativeToken.chain.chainID ?? 0, walletAddress: nativeToken.address)
```

Comment:
Didn't got the idea. We're calling `fetchTokensBalance` that returns array of 1inch tokens, but after mapping it into array of CoinMeta that does not have balance property. Do we really fetch balances here?

Reply:
May be just wrong `fetchTokensBalance` function name

Reply:
This method discovers the tokens the user has on its address

```
static func fetch1InchsTokensBalance(chain: String, address: String) -> String {
        return "\(vultisigApiProxy)/1inch/balance/v1.2/\(chain)/balances/\(address)"
    }
 ```
 
 so it returns something like:
 
 ```
 {
 "0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee":0,
 "0x85f17cf997934a597031b2e18a9ab6ebd4b9f6a4":18371273
 }
```

Then I filter by the contracts they have balances to consider it a token

Reply:
I also renamed the function